### DISCUSSION TITLE
GoodbyeDPI UI v1.2.1

### DISCUSSION THREAD

**Discussion by Storik4pro** [Storik4pro/goodbyeDPI-UI#24] [Announcements]
## Что нового в этом обновлении?

- 🚀 Был совершен полный переход на Qt для повышения производительности и упрощения использования приложения. Благодаря этому удалось решить проблему с задержкой при открытии настроек (#4) и проблему с эффектом Mica (#18)
- ✨ Теперь вы можете открыть параметры компонента в отдельном окне [нажмите ПКМ по разделу параметров -> `"открыть в новом окне"`]
- ✨ Были внесены изменения в меню трея. Теперь изменять компонент по умолчанию, запускать/останавливать процесс и просматривать его вывод можно не открывая основное приложение (#17)
- ✨ Были добавлены уведомления о наличии обновлений для компонентов. Вы можете в любой момент отключить в меню `"Система"` -> `"Уведомления на рабочем столе"` (#7)
- ✨ Теперь приложение будет удалять ненужные данные после обновления, очищая тем самым от 80 до 500 мб ненужных файлов
- 💎 Теперь меню "Работа с пресетами" показывает текущий файл конфигурации.
- 🔼 Версия zapret по умолчанию была обновлена до v66

## БЕТА - версии функций

- 🛠 Теперь в качестве утилиты подбора параметров будет использоваться тестовая версия [GoodCheck by Ori на Go](https://ntc.party/t/goodcheck-%D0%B1%D0%BB%D0%BE%D0%BA%D1%87%D0%B5%D0%BA-%D1%81%D0%BA%D1%80%D0%B8%D0%BF%D1%82-%D0%B4%D0%BB%D1%8F-goodbyedpi-zapret-byedpi/10880/1012). 
- 🛠 Теперь, с помощью утилиты подбора параметров запуска, можно подобрать параметры для zapret

## Прочее 

- 🚀 Был изменен способ упаковки приложения, благодаря чему улучшилась производительность и исправлена проблема, из-за которой антивирус мог ложно определить приложение как троян (#21). 
**Обратите внимание.** Антивирусы все ещё могут пытаться удалить архив или его части из-за наличия zapret и windivert в архиве. Так же антивирус может запретить добавление приложения в автозапуск.

## Список исправлений

- Теперь ошибка `отказано в доступе: "logs"` не будет появляться при первом и последующих запусках (#19) 
- Теперь пользовательские конфигурации будут загружаться корректно при автозапуске (#15)
- Была исправлена ошибка из-за которой остановка процесса после изменение компонента по умолчанию во время работы компонента на Windows 10 младше версии 1809, могла вызвать ошибку
- Если компонент раннее был запущен, то при попытке его удалить вы больше не увидите ошибку `ERR_PERMISSION_DENIED`
- Была исправлена проблема в механизме обновления, из-за которой Update Assistant оставался запущенным после завершения установки

## Известные проблемы

- Запуск приложения может занимать больше времени, в зависимости от установленных компонентов и скорости сети.

## What's new in this update?

- 🚀 The application has fully transitioned to Qt to improve performance and simplify usage. This resolves the delay issue when opening settings (#4) and the Mica effect issue (#18).
- ✨ You can now open component settings in a separate window [right-click on the settings section -> `"View in new window"`].
- ✨ The system tray menu has been updated. Now, you can change the default component, start/stop the process, and view its output without opening the main application (#17).
- ✨ Update notifications for components have been added. You can disable them at any time in the `"System"` -> `"Desktop notifications"` menu (#7).
- ✨ The application now removes unnecessary files after updates, freeing between 80 to 500 MB of storage.
- 💎 The "Preset Management" menu now shows the current configuration file.
- 🔼 The default version of zapret has been updated to v66.

## BETA features

- 🛠 The test version of [GoodCheck by Ori in Go](https://ntc.party/t/goodcheck-%D0%B1%D0%BB%D0%BE%D0%BA%D1%87%D0%B5%D0%BA-%D1%81%D0%BA%D1%80%D0%B8%D0%BF%D1%82-%D0%B4%D0%BB%D1%8F-goodbyedpi-zapret-byedpi/10880/1012) will now be used as the parameter selection utility.
- 🛠 You can now use the parameter selection utility to configure zapret parameters.

## Other

- 🚀 The packaging method has been changed, improving performance and resolving an issue where antivirus software could mistakenly detect the app as a trojan (#21).  
**Note:** Antivirus software may still attempt to delete the archive or its parts due to the presence of zapret and windivert in the archive. It may also prevent adding the app to autostart.

## List of fixes

- The `Access denied: "logs"` error will no longer appear on the first or subsequent launches (#19).
- User configurations will now load correctly on autostart (#15).
- Fixed an issue where stopping the process after changing the default component while running on Windows 10 versions below 1809 could cause an error.
- If a component was previously running, you will no longer see the `ERR_PERMISSION_DENIED` error when attempting to delete it.
- Fixed an update mechanism issue where the Update Assistant would remain running after installation completion.

## Known issues

- The application launch time may vary based on installed components and network speed.

<hr /><em>This discussion was created from the release <a href='https://github.com/Storik4pro/goodbyeDPI-UI/releases/tag/1.2.1'>GoodbyeDPI UI v1.2.1</a>.</em>

**Comment by xleb-normal** [comment]
Йоу очень крутое обновление, UI мега хорошо выглядит!

**Comment by MaysStudio** [comment]
Не работает
